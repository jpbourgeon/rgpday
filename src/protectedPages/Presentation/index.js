import React from 'react'
import loadable from '@loadable/component'
import { withStyles } from '@material-ui/core/styles'

const styles = () => ({
  '@global': {
    '.grecaptcha-badge': {
      visibility: 'hidden'
    }
  }
})

const AsyncPresentation = loadable(props => import(`../../presentations/${props.presentationId}`))

const Presentation = (props) => {
  const { presentationId } = props.config
  if (presentationId !== null) {
    return (<AsyncPresentation presentationId={presentationId} />)
  } else {
    return null
  }
}

export default withStyles(styles)(Presentation)
