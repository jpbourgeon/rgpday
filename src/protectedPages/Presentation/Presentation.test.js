/* global describe, it, expect */
import React from 'react'
import renderer from 'react-test-renderer'
import Presentation from './'
import jssSerializer from 'src/utils/jssSnapshotSerializer'

expect.addSnapshotSerializer(jssSerializer)

describe('The Presentation component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Presentation config={{ scenarioId: '2019-INGE-01' }} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
