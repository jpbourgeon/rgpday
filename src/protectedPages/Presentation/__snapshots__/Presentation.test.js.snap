// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Presentation component should render correctly 1`] = `null`;
